pkg/controller/pravegacluster/upgrade.go

```diff
@@ -71,6 +71,14 @@ func (r *ReconcilePravegaCluster) syncClusterVersion(p *pravegav1alpha1.PravegaC
 		if err != nil {
 			log.Printf("error syncing cluster version, upgrade failed. %v", err)
 			p.Status.SetErrorConditionTrue("UpgradeFailed", err.Error())
+			// emit an event for Upgrade Failure
+			message := fmt.Sprintf("Error Upgrading from version %v to %v. %v", p.Status.CurrentVersion, p.Status.TargetVersion, err.Error())
+			reason := "UpgradeError"
```

Comment:
It would be better to put reason in status.go file as other reasons are specified there.